### ISSUE TITLE
Program, universe polymorphism and ssr rewrite.

### ISSUE THREAD

**Issue by arthuraa** [coq/coq#11988]
#### Description of the problem

```coq
From Coq Require Import ssreflect.
Require Import Coq.Program.Tactics.

Obligation Tactic := idtac.

Set Universe Polymorphism.

Section Foo.

Universe u.

Variable T : Type@{u}.
Variable op : T -> T -> T.
Hypothesis opC : forall x y, op x y = op y x.

Program Definition foo@{} x y : {z : T & z = op x y} :=
  existT _ (op y x) _.

Next Obligation.
move=> /= x y.
(* The pattern seems to introduce a spurious universe, which clashes with
the universe annotation in foo. *)
by rewrite -[RHS]opC.
Fail Qed.

Restart.
move=> /= x y.
(* Without the pattern, the no extra universe is introduced. *)
by rewrite -(opC x y).
Qed.

End Foo.

```

#### Coq Version

8.11

**Comment by ejgallego** [comment]
This is a bug, fixed by #11902

**Comment by SkySkimmer** [comment]
Duplicate https://github.com/coq/coq/issues/11766
